src/jit/decomposelongs.cpp

```diff
@@ -1380,9 +1366,30 @@ GenTree* DecomposeLongs::DecomposeRotate(LIR::Use& use)
     }
     else
     {
-        Range().Remove(gtLong);
-        loOp1 = RepresentOpAsLocalVar(loOp1, gtLong, &gtLong->gtOp.gtOp1);
-        hiOp1 = RepresentOpAsLocalVar(hiOp1, gtLong, &gtLong->gtOp.gtOp2);
+        GenTree* loResult;
+        GenTree* hiResult;
+
+        if (count > 32)
+        {
+            // If count > 32, we swap hi and lo, and subtract 32 from count
+            hiOp1 = gtLong->gtGetOp1();
+            loOp1 = gtLong->gtGetOp2();
+
+            Range().Remove(gtLong);
+            loOp1 = RepresentOpAsLocalVar(loOp1, gtLong, &gtLong->gtOp.gtOp2);
+            hiOp1 = RepresentOpAsLocalVar(hiOp1, gtLong, &gtLong->gtOp.gtOp1);
+
+            count -= 32;
+        }
+        else
+        {
+            loOp1 = gtLong->gtGetOp1();
+            hiOp1 = gtLong->gtGetOp2();
+
+            Range().Remove(gtLong);
+            loOp1 = RepresentOpAsLocalVar(loOp1, gtLong, &gtLong->gtOp.gtOp1);
+            hiOp1 = RepresentOpAsLocalVar(hiOp1, gtLong, &gtLong->gtOp.gtOp2);
+        }
```

Comment:
As discussed offline two things about lines 1397-1406 :1) both if and else blocks are doing the same but in different order and 2) since count is subtracted 32 if > 32, it will always be < 32.